src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2609,7 +2609,7 @@ private BoundStatement BindReturn(ReturnStatementSyntax syntax, DiagnosticBag di
 
             if (hasErrors)
             {
-                return new BoundReturnStatement(syntax, refKind, arg, hasErrors: true);
+                return new BoundReturnStatement(syntax, sigRefKind, arg, hasErrors: true);
```

Comment:
>sigRefKind [](start = 56, length = 10)

If we are planning to expose that through IOperation shouldn't this reflect what we have in source? I.e. if there was no ```ref``` in return statement in source, we should use ```RefKind.None``` regardless what the method signature says. And the other way around, if signature doesn't have ```ref``` and the return statement has ```ref```, we should use ```refKind```.

Reply:
This is now resolved.